### ISSUE TITLE
Avoid showing blank covers (if book readable)

### ISSUE THREAD

**Issue by BrittanyBunk** [internetarchive/openlibrary#2486]
Also related: #2447 #2132

- A lot of book covers have nothing on it, which make it difficult to know what the book's about in the carousels. It would be good to have a bot that finds these covers and then adds one that has a description on it - which would be through the preview 99% of the time.
- Many works have no covers, yet have one edition with a book cover. It would be great to have an automatic setting where one cover comes into existence and no top cover's there, then it takes the first cover added in.

**Comment by cdrini** [comment]
- For your first point, does this solve the issue? https://github.com/internetarchive/openlibrary/issues/2247
- For your second, you're suggesting that there be a way for the work to automatically display an edition's cover if one exists, correct?

**Comment by BrittanyBunk** [comment]
Idk if they're related, as that one seems to be a glitch. Mine looks like 
![Opera Snapshot_2019-10-12_150824_openlibrary org](https://user-images.githubusercontent.com/55302685/66708236-45e71380-ed02-11e9-99b2-d69383b5f03c.png)
![Untitled](https://user-images.githubusercontent.com/55302685/66708237-45e71380-ed02-11e9-8d9e-e2c3b3b3a124.png).

The second point's to automatically add a cover to a work the second one of the editions gets one if it doesn't have one already.

**Comment by mekarpeles** [comment]
If the carousel item or search result work is readable / borrowable / has daisy / waitlist ... i.e. if we've identified the work has an archive.org edition, we can use it's `ocaid` to fallback to an archive.org cover `https://archive.org/services/img/{identifier}`. I think @leadsongdog would probably agree we should wrap this archive.org url within covers.openlibrary.org (i.e. use a proxy) so patrons don't have a bad experience in areas where archive.org may be blocked by openlibrary.org isn't.

**Comment by mekarpeles** [comment]
Also related: #2447 #2132

**Comment by mekarpeles** [comment]
@ignatiusdork is working on this issue -- closing this specific issue as a duplicate of:
- #2550
- #2514
